sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -21,7 +21,8 @@ def _tosequence(X):
         return tosequence(X)
 
 
-class DictVectorizer(BaseEstimator, TransformerMixin):
+class DictVectorizer(BaseEstimator, TransformerMixin,
+                     NonRectangularInputMixin):
```

Comment:
should we start adding the mixins to the left? [related](https://stackoverflow.com/questions/10018757/how-does-the-order-of-mixins-affect-the-derived-class)

I kinda had to in one case when I was playing with the feature names using xarray.

Reply:
I think that should be a separate discussion.